### PR TITLE
feat: Add scaler and update repo (SC-16792)

### PR DESCRIPTION
**Pull request by JGcarv**

### CODE REVIEW

File: contracts/Migrator.sol [hunk 1]
```diff
@@ -7,6 +7,8 @@ import { IERC20Like } from "./interfaces/Interfaces.sol";
 
 contract Migrator {
 
+    uint256 public constant SCALER = 10;
```
Comment (0xfarhaan) [inline contracts/Migrator.sol:10]: let's make this immutable and set in the constructor please

Comment (0xfarhaan) [reply contracts/Migrator.sol:10]: let's rename to TOKEN_SPLIT_SCALAR, scalar alone is too generic (also note the spelling with an A)

File: contracts/Migrator.sol [hunk 2]
```diff
@@ -23,8 +25,11 @@ contract Migrator {
 
     function migrate(address owner_, uint256 amount_) public {
         require(amount_ != uint256(0),                                              "M:M:ZERO_AMOUNT");
+
+        uint256 outputAmount = amount_ * SCALER;
```
Comment (0xfarhaan) [inline contracts/Migrator.sol:29]: why not just calculate in the require statement as it's used once?

File: contracts/test/Migrator.t.sol [hunk 3]
```diff
@@ -1,22 +1,14 @@
...
-contract SomeAccount {
-
-    function approve(address token_, address spender_, uint256 amount_) external {
-        IERC20(token_).approve(spender_, amount_);
-    }
-
-}
-
-contract MigratorConstructorTest is TestUtils {
+contract MigratorConstructorTest is Test {
```
Comment (0xfarhaan) [inline contracts/test/Migrator.t.sol:11]: Let's move this test folder and files out to have the repo reflect our current repos

File: contracts/test/Migrator.t.sol [hunk 4]
```diff
@@ -1,22 +1,14 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 pragma solidity 0.8.7;
 
-import { IERC20 } from "../../modules/erc20/contracts/interfaces/IERC20.sol";
+import { Test } from "../../modules/forge-std/src/Test.sol";
 
-import { TestUtils } from "../../modules/contract-test-utils/contracts/test.sol";
+import { IERC20 }    from "../../modules/erc20/contracts/interfaces/IERC20.sol";
```
Comment (0xfarhaan) [inline contracts/test/Migrator.t.sol:6]: import not used

File: contracts/Migrator.sol [hunk 5]
```diff
@@ -7,6 +7,8 @@ import { IERC20Like } from "./interfaces/Interfaces.sol";
 
 contract Migrator {
```
Comment (0xfarhaan) [inline contracts/Migrator.sol:8]: Let's have this contract inherit the interface and update the IMigrator interface to include the new scalar parm

**Review by 0xfarhaan — CHANGES_REQUESTED** [review]

File: tests/Migrator.t.sol [hunk 6]
```diff
@@ -1,47 +1,41 @@
...
 
     function test_constructor() external {
         MockERC20 oldToken = new MockERC20("Old Token", "OT", 18);
         MockERC20 newToken = new MockERC20("New Token", "NT", 18);
 
-        Migrator migrator = new Migrator(address(oldToken), address(newToken));
+        Migrator migrator = new Migrator(address(oldToken), address(newToken), 1);
 
         assertEq(migrator.oldToken(), address(oldToken));
         assertEq(migrator.newToken(), address(newToken));
```
Comment (0xfarhaan) [inline tests/Migrator.t.sol:27]: we should assert the scalar

File: contracts/Migrator.sol [hunk 7]
```diff
@@ -5,26 +5,33 @@ import { ERC20Helper } from "../modules/erc20-helper/src/ERC20Helper.sol";
...
 
-contract Migrator {
+import { IMigrator } from "./interfaces/IMigrator.sol";
 
-    address public immutable newToken;
-    address public immutable oldToken;
+contract Migrator is IMigrator {
 
-    constructor(address oldToken_, address newToken_) {
+    uint256 public immutable override TOKEN_SPLIT_SCALAR;
```
Comment (0xfarhaan) [inline contracts/Migrator.sol:12]: nit if not constant we should lowercase and order based on type i.e `uint256` after `address`

**Review by 0xfarhaan — COMMENTED** [review]
minor comments

File: contracts/Migrator.sol [hunk 8]
```diff
@@ -5,26 +5,33 @@ import { ERC20Helper } from "../modules/erc20-helper/src/ERC20Helper.sol";
...
+
+    address public immutable override newToken;
+    address public immutable override oldToken;
+
+    constructor(address oldToken_, address newToken_, uint256 scalar_) {
```
Comment (vbidin) [inline contracts/Migrator.sol:17]: Can add a `require` for the scalar to be greater than zero.

**Review by 0xfarhaan — APPROVED** [review]

**Pull request merged**